src/app/views/marketplace/modals/answer-modal.coffee

```diff
@@ -0,0 +1,16 @@
+angular.module 'mnoEnterpriseAngular'
+.controller('AnswerModal', ($scope, $uibModalInstance, question) ->
+
+  $scope.answerText = ''
+  $scope.questionText = question.text
+
+  # Close the current modal
+  $scope.closeModal = ->
+    $uibModalInstance.dismiss()
+
+  $scope.submitAnswer = ->
+    question.answers.push({text: $scope.answerText})
+    $uibModalInstance.dismiss()
```

Comment:
Please do not use `dismiss` prefer `$uibModalInstance.close($scope.answerText)` and process the object in the modal result. (cf. https://angular-ui.github.io/bootstrap/#/modal)